### PR TITLE
initialize direction metadata at begining of pipeline

### PR DESCRIPTION
**Pull request by kamleshbhalui**
Read direction only once in a pipeline and input port and direction is same throughout

### CODE REVIEW

File: backends/dpdk/dpdkMetadata.cpp [hunk 1]
```diff
@@ -60,51 +64,29 @@ IR::DpdkExternDeclaration* DirectionToRegRead::addRegDeclInstance(cstring instan
...
-    }
+    IR::IndexedVector<IR::DpdkAsmStatement> newStmts;
+    newStmts.insert(newStmts.begin(), *stmts.begin());
+    // insert direction read after rx statement
+    auto dirMeta = new IR::Member(new IR::PathExpression(IR::ID("m")),
+                                  IR::ID("pna_main_input_metadata_direction"));
+    auto inputPort = new IR::Member(new IR::PathExpression(IR::ID("m")),
+                                    IR::ID("pna_main_input_metadata_input_port"));
+    auto reads = new IR::DpdkRegisterReadStatement(dirMeta, registerInstanceName, inputPort);
+    newStmts.insert(newStmts.begin() + 1, reads);
```
Comment (usha1830) [inline backends/dpdk/dpdkMetadata.cpp:88]: Add a few comments why we are adding at specific index

File: backends/dpdk/dpdkMetadata.cpp [hunk 2]
```diff
@@ -35,14 +35,18 @@ void DirectionToRegRead::uniqueNames(IR::DpdkAsmProgram* p) {
...
+    p->apply(*IsDirUsed);
     uniqueNames(p);
     p->externDeclarations.push_back(addRegDeclInstance(registerInstanceName));
-    IR::IndexedVector<IR::DpdkAsmStatement> stmts;
-    for (auto stmt : p->statements) {
-        auto stmtList = stmt->to<IR::DpdkListStatement>()->clone();
-        stmts.push_back(replaceDirection(stmtList));
+    if (is_direction_used) {
+        IR::IndexedVector<IR::DpdkAsmStatement> stmts;
+        stmts.push_back(new IR::DpdkListStatement(replaceDirectionWithRegRead(
```
Comment (usha1830) [inline backends/dpdk/dpdkMetadata.cpp:46]: Name of this function can be changed to insertRegReadForDirection or something similar

**Review by usha1830 — APPROVED** [review]
Looks fine

**Review by syogender — APPROVED** [review]
Looks Good

**Pull request closed without merging**